File: java/TestLoginHelper.java
package workbooks_app.client_lib.java;

import java.util.HashMap;

/** 
 *	Login wrapper for Workbooks for API test purposes. This version uses an API Key to
 *  authenticate which is the recommended approach unless you are running under the 
 *  Process Engine which will set up a session for you automatically without requiring
 *  an API key.
 * 
 * 	License: www.workbooks.com/mit_license
 * 	Last commit $Id: TestLoginHelper.java 65931 2025-03-06 15:19:28Z jkay $
 */

public class TestLoginHelper {
	
	HashMap<String, Object> params = new HashMap<String,Object>();
	String service = System.getenv("WB_SERVICE").isEmpty() ? "http://localhost:3000" : System.getenv("WB_SERVICE");
	String application_name = "java_test_client";
	String user_agent = "java_test_client/0.1";
	String api_key ="01234-56789-01234-56789-01234-56789-01234-56789";
	boolean verify_peer = false;
	
	public WorkbooksApi workbooks = null;
	HashMap<String, Object> logObjects = new HashMap<String, Object>();
	
	public WorkbooksApi testLogin() {
		if (service != null) {
			params.put("service", service);
		}
		params.put("application_name", application_name);
		params.put("user_agent", user_agent);
		params.put("verify_peer", verify_peer);
		if (api_key != null && api_key != "") {
			params.put("api_key", api_key);
		}
		try {
			workbooks = new WorkbooksApi(params);
			workbooks.log("Logged in with these params: ", new Object[] {params} );
		} catch(Exception e) {
			workbooks.log("Error while creating the Workbooks API object: ", new Object[] {e}, "error", WorkbooksApi.DEFAULT_LOG_LIMIT);
			e.printStackTrace();
		}
		
		return workbooks;
	}
	
	public void testExit(WorkbooksApi workbooks, int exitcode) {
		
		if (exitcode == 0) {
			workbooks.log("Script exited", new Object[] {"OK"}, "info", WorkbooksApi.DEFAULT_LOG_LIMIT);
		} else {
			workbooks.log("Script exited with error", new Object[] {Integer.toString(exitcode)}, "error", WorkbooksApi.DEFAULT_LOG_LIMIT);
		}
		System.exit(exitcode);
		
	}
	
	public static void main(String[] args) {
		TestLoginHelper testLoginHelper = new TestLoginHelper();
		testLoginHelper.testLogin();
	}
}

File: csharp/TestLoginHelper.cs
 /**
  * 	License: www.workbooks.com/mit_license
  * 	Last commit $Id: TestLoginHelper.cs 63933 2024-09-03 14:06:12Z jmonahan $
  *
  * Login wrapper for Workbooks for API test purposes. This version uses an API Key to
  * authenticate which is the recommended approach unless you are running under the
  * Process Engine which will set up a session for you automatically without requiring
  * an API key.
  */
using System;
using System.IO;
using System.Net;
using System.Text;
using System.Collections.Generic;
using System.Collections;
using System.Runtime.Serialization.Json;
using System.Web;
using WorkbooksApiApplication;

namespace WorkbooksApiApplication
{
	public class TestLoginHelper
	{
		Dictionary<string, object> loginParams = new Dictionary<string, object>();
    string service = "http://localhost:3000";
		string application_name = "csharp_test_client";
		string user_agent = "csharp_test_client/0.1";
		string api_key ="01234-56789-01234-56789-01234-56789-01234-56789";
		bool verify_peer = false;

		public WorkbooksApi workbooks = null;

		public WorkbooksApi testLogin() {

      // allow the server and API key to be overridden with environment variables
      string service_env = Environment.GetEnvironmentVariable("WB_SERVICE");
      string api_key_env = Environment.GetEnvironmentVariable("WB_API_KEY");

      if (service_env != null) {
        service = service_env;
      }

      if (api_key_env != null) {
        api_key = api_key_env;
      }

			if (service != null) {
				loginParams.Add("service", service);
			}
			loginParams.Add("application_name", application_name);
			loginParams.Add("user_agent", user_agent);
			loginParams.Add("verify_peer", verify_peer);
			if (api_key != null && api_key != "") {
				loginParams.Add("api_key", api_key);
			}
			try {
				workbooks = new WorkbooksApi(loginParams);
				Console.WriteLine("Workbooks: " + workbooks.Service);
				//				workbooks.log("Logged in with these loginParams: ", new Object[] {loginParams} );
			} catch(Exception e) {
				workbooks.log("Error while creating the Workbooks API object: ", new Object[] {e}, "error", WorkbooksApi.DEFAULT_LOG_LIMIT);
				Console.WriteLine ("Stacktrace: " + e.StackTrace);
			}
			return workbooks;
		}

		public void testExit(WorkbooksApi workbooks, int exitcode) {

			if (exitcode == 0) {
        //Console.WriteLine ("Scripted exited OK");
        workbooks.log("Script exited", new Object[] {"OK"}, "info");
			} else {
        //Console.WriteLine ("Scripted exited with error");
        workbooks.log("Script exited with error", new Object[] {(int)exitcode}, "error");
			}
			System.Environment.Exit(exitcode);
		}

	}
}

